packages/adblocker-benchmarks/blockers/tsurlfilter.js

```diff
@@ -39,18 +46,33 @@ module.exports = class TSUrlFilter {
     // The first argument is the list ID and must be unique
     const list = new StringRuleList(1, rawLists);
     const ruleStorage = new RuleStorage([ list ]);
-    return new TSUrlFilter(new Engine(ruleStorage));
+    const engine = TSUrlFilter.hostsOnly ?
+                     new DnsEngine(ruleStorage) :
+                     new Engine(ruleStorage);
+    return new TSUrlFilter(engine);
   }
 
   constructor(engine) {
     this.engine = engine;
   }
 
   match({ url, frameUrl, type }) {
+    const result = TSUrlFilter.hostsOnly ?
+                     this.matchHostname(url) :
+                     this.matchRequest({ url, frameUrl, type });
+    return result !== null && !result.whitelist;
```

Comment:
Something worth noting is that the DNS engine also supports whitelisting rules. This is also true of other content blockers like uBlock Origin. But we have no whitelisting rules in our [`hosts.txt`](https://github.com/cliqz-oss/adblocker/blob/master/packages/adblocker-benchmarks/hosts.txt) file here.

Reply:
Well, it also supports a little different subset of rules & modifiers, the syntax is explained here:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists